controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -161,10 +181,14 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 // GetIntegrationPipelineRunStatus checks the Tekton results for a given PipelineRun and returns status of test.
 func GetIntegrationPipelineRunStatus(adapterClient client.Client, ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (intgteststat.IntegrationTestStatus, string, error) {
 	// Check if the pipelineRun finished from the condition of status
-	if !h.HasPipelineRunFinished(pipelineRun) {
+	if !h.HasPipelineRunFinished(pipelineRun) && pipelineRun.GetDeletionTimestamp() == nil {
 		return intgteststat.IntegrationTestStatusInProgress, fmt.Sprintf("Integration test is running as pipeline run '%s'", pipelineRun.Name), nil
 	}
 
+	if !h.HasPipelineRunFinished(pipelineRun) && pipelineRun.GetDeletionTimestamp() != nil {
```

Comment:
it may deserve comment what non-nil DeletionTimestamp means here

Reply:
Added a comment

Reply:
looking at the code it could be
```
if PLR hasnt fnished:
   if deletion_timestamp:
       return StatusDeleted
   else:
       return  StatusRunning
```

instead of separate conditions with `&&`